Tests/SwiftDocCTests/Infrastructure/DocumentationContext/DocumentationContextTests.swift

```diff
@@ -644,15 +644,25 @@ class DocumentationContextTests: XCTestCase {
     
     func testDetectsReferenceCollision() throws {
         let (_, context) = try testBundleAndContext(named: "TestBundleWithDupe")
-        
-        XCTAssert(
-            context.problems.contains { problem in
-                problem.diagnostic.identifier == "org.swift.docc.DuplicateReference"
-                    && problem.diagnostic.localizedSummary == "Redeclaration of 'TestTutorial.tutorial'; this file will be skipped"
-            }
-        )
+
+        let problemWithDuplicate = context.problems.filter{ $0.diagnostic.identifier == "org.swift.docc.DuplicateReference" }
+
+        XCTAssertEqual(problemWithDuplicate.count, 1)
+
+        XCTAssertEqual(problemWithDuplicate[0].diagnostic.localizedSummary, "Redeclaration of 'TestTutorial.tutorial'; this file will be skipped")
+
     }
     
+    func testDetectsMultipleMDfilesWithSameName() throws {
+        let (_, context) = try testBundleAndContext(named: "TestBundleWithDupMD")
+
+        let problemWithDuplicateReference = context.problems.filter { $0.diagnostic.identifier == "org.swift.docc.DuplicateReference" }
+
+        XCTAssertEqual(problemWithDuplicateReference.count, 2)
+
+        XCTAssertEqual(problemWithDuplicateReference[0].diagnostic.localizedSummary,  "Redeclaration of \'overview.md\'; this file will be skipped")
```

Comment:
Same here.

What is the second diagnostic? Should we assert its contents as well?

Reply:
it is the same, since we are checking the count of "org.swift.docc.DuplicateReference" to be 2, i skipped adding checking the localizedSummary.